### PR TITLE
MINOR: Add test cases for delays in consumer rebalance listener

### PR DESCRIPTION
**Pull request by hachikuji**

### CODE REVIEW

File: core/src/test/scala/integration/kafka/api/PlaintextConsumerTest.scala [hunk 1]
```diff
@@ -83,6 +85,76 @@ class PlaintextConsumerTest extends BaseConsumerTest {
...
+    var commitCompleted = false
+    var committedPosition: Long = -1
+
+    val listener = new TestConsumerReassignmentListener {
+      override def onPartitionsRevoked(partitions: util.Collection[TopicPartition]): Unit = {
+        if (callsToRevoked > 0) {
+          // on the second rebalance (after we have joined the group initially), sleep longer
+          // than session timeout and then try a commit. We should still be in the group,
+          // so the commit should succeed
+          Utils.sleep(1500)
```
Comment (ijuma) [inline core/src/test/scala/integration/kafka/api/PlaintextConsumerTest.scala:101]: I guess we can't avoid this `sleep` here right?

**Comment by ijuma** [comment]
Thanks for the PR @hachikuji, LGTM. Did you test that these fail without your fix for KAFKA-4160?

File: core/src/test/scala/integration/kafka/api/PlaintextConsumerTest.scala (hunk 1 above)
Comment (hachikuji) [reply core/src/test/scala/integration/kafka/api/PlaintextConsumerTest.scala:101]: Sigh, I couldn't think of another way to do it.

**Comment by hachikuji** [comment]
@ijuma Thanks for reviewing! I did verify that the revocation test fails prior to 4160. The other test passes since we were already covering that case, but I couldn't find a test case for it.

**Comment by ijuma** [comment]
Merged to trunk.

**Pull request closed without merging**